src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
```

Comment:
In fact, since we cannot (at the moment) use `Batch` API on `DocumentFragment` these markers are for sure not added to the document, so I hesitate whether we should add them to the document at all.